### ISSUE TITLE
Remove all other hop hints from invoices that need a channel open

### ISSUE THREAD

**Issue by JssDWt** [breez/breez-sdk#539]
If an open channel is needed, because inbound liquidity is too low to receive a payment, a hop hint to the LSP is added to the invoice. If the user's node has channels to other nodes however, these hop hints will still be in the created invoice. That means a multipart payment could route some parts over the lsp, but some parts over the other connected nodes. Routing some parts over the other connected nodes breaks the lsp model, because the assumption of the lsp currently is that all parts of the payment are routed over the lsp.

TL;DR if a channel open is needed, _only_ include a hop hint to the lsp in the invoice.
